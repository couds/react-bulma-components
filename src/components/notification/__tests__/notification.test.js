import React from 'react';
import renderer from 'react-test-renderer';
import { Notification } from '..';
import CONSTANTS from '../../../constants';

describe('Notification component', () => {
  it('should exist', () => {
    expect(Notification).toMatchSnapshot();
  });
  it('should have notification classname', () => {
    const component = renderer.create(
      <Notification>
        <img alt="placeholder" src="http://bulma.io/images/placeholders/128x128.png" />
      </Notification>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const component = renderer.create(
      <Notification className="other-class this-is-a-test">
        <p>Default</p>
      </Notification>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const component = renderer.create(
      <Notification style={{ height: 250 }}>
        <p>Default</p>
      </Notification>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should render as Section', () => {
    const component = renderer.create(
      <Notification renderAs="section">
        <p>Default</p>
      </Notification>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  Object.values(CONSTANTS.COLORS).map(color =>
    it(`Should use use color ${color}`, () => {
      const component = renderer.create(
        <Notification color={color}>
          <p>Default</p>
        </Notification>
      );
      expect(component.toJSON()).toMatchSnapshot();
    })
  );
});
