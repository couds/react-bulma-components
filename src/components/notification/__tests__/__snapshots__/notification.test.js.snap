// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component Should use use color black 1`] = `
<div
  className="notification is-black"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color danger 1`] = `
<div
  className="notification is-danger"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color dark 1`] = `
<div
  className="notification is-dark"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color info 1`] = `
<div
  className="notification is-info"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color light 1`] = `
<div
  className="notification is-light"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color link 1`] = `
<div
  className="notification is-link"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color null 1`] = `
<div
  className="notification"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color primary 1`] = `
<div
  className="notification is-primary"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color success 1`] = `
<div
  className="notification is-success"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color warning 1`] = `
<div
  className="notification is-warning"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component Should use use color white 1`] = `
<div
  className="notification is-white"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component should concat classname in props with classname 1`] = `
<div
  className="notification other-class this-is-a-test"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Notification component should exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "backgroundColor": null,
    "children": null,
    "className": "",
    "clearfix": false,
    "clipped": false,
    "color": null,
    "hidden": false,
    "invisible": false,
    "italic": false,
    "marginless": false,
    "overlay": false,
    "paddingless": false,
    "pull": null,
    "radiusless": false,
    "renderAs": "div",
    "responsive": undefined,
    "shadowless": false,
    "style": Object {},
    "textAlignment": null,
    "textColor": null,
    "textSize": null,
    "textTransform": null,
    "textWeight": null,
    "unselectable": false,
  },
  "propTypes": Object {
    "backgroundColor": [Function],
    "children": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "color": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "renderAs": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
  },
  "render": [Function],
}
`;

exports[`Notification component should have notification classname 1`] = `
<div
  className="notification"
  style={Object {}}
>
  <img
    alt="placeholder"
    src="http://bulma.io/images/placeholders/128x128.png"
  />
</div>
`;

exports[`Notification component should render as Section 1`] = `
<section
  className="notification"
  style={Object {}}
>
  <p>
    Default
  </p>
</section>
`;

exports[`Notification component should use inline styles 1`] = `
<div
  className="notification"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</div>
`;
